CMake/PickyWarnings.cmake

```diff
@@ -89,6 +88,14 @@ if(PICKY_COMPILER)
       -Wwrite-strings                      # clang  1.0  gcc  1.4
     )
 
+    if(NOT (ENABLE_DEBUG OR ENABLE_CURLDEBUG))
+      # If building DEBUG images, do not error on -Winline as in may not
+      # be supported
+      list(APPEND WPICKY_ENABLE
+        -Winline                           # clang  1.0  gcc  1.0
+      )
+    endif()
+
```

Comment:
@vszakats we have discussed this option before. It is not used when configure sets pick options. Its value and use is questionable. Should we just never use it?

Reply:
Building with curl debug functionality + memory tracing + picky warnings
and `CURL_WERROR` explicitly enabled looks like an atypical build case
to me. `ENABLE_DEBUG` is meant to debug curl itself and basically removes
security.

Was this intented? If not, this might be a legit (though somewhat accidental)
warning sign, and possible replace `ENABLE_DEBUG=ON` with 
`-DCMAKE_BUILD_TYPE=Debug`, which creates a secure/production curl
with debug info enabled.

In case it was intented and we want to make it pass, it's better to delete
this flag completely, rather then adding more conditionals.